File: 短剧/backend/app/schemas/composition.py
"""
合成任务 Pydantic 模型（请求/响应 Schema）。
"""

from typing import Any, Optional

from pydantic import BaseModel, Field


class CompositionRequest(BaseModel):
    """合成请求体。"""

    script_id: Optional[str] = Field(default=None, description="剧本 ID")
    project_id: Optional[str] = Field(default=None, description="关联项目 ID")
    episode_no: int = Field(..., ge=1, description="集数")
    storyboard_ids: list[str] = Field(..., min_length=1, description="要合成的分镜 ID 列表")


class CompositionResponse(BaseModel):
    """合成任务响应体。"""

    id: str
    script_id: Optional[str] = None
    episode_no: int
    status: str
    output_url: Optional[str] = None
    error_message: Optional[str] = None
    progress: Optional[str] = None

File: 短剧/frontend/src/api/composition.ts
/**
 * 合成任务相关 API 调用封装。
 */

import apiClient from './index'

/** 合成任务响应体。 */
export interface CompositionResponse {
  id: string
  script_id: string | null
  episode_no: number
  status: 'pending' | 'processing' | 'success' | 'failed'
  output_url: string | null
  error_message: string | null
  progress: string | null
}

/** 合成请求体。 */
export interface CompositionRequest {
  script_id?: string
  project_id?: string
  episode_no: number
  storyboard_ids: string[]
}

export const compositionApi = {
  /** 提交合成任务。 */
  generateEpisode: (data: CompositionRequest) =>
    apiClient.post<CompositionResponse>('/composition/generate-episode', data),

  /** 查询合成任务状态。 */
  getTask: (taskId: string) =>
    apiClient.get<CompositionResponse>(`/composition/task/${taskId}`),

  /** 获取所有合成任务，可选按项目过滤。 */
  list: (projectId?: string) =>
    apiClient.get<CompositionResponse[]>('/composition', {
      params: projectId ? { project_id: projectId } : {},
    }),
}
